tests/tui.rs

```diff
@@ -19,70 +19,103 @@
 
 use dft::app::state::initialize;
 use dft::app::{App, AppEvent};
+use dft::execution::ExecutionContext;
 use ratatui::crossterm::event;
-use tempfile::tempdir;
+use tempfile::{tempdir, TempDir};
 
 #[tokio::test]
-async fn run_app_with_no_args() {
-    let config_path = tempdir().unwrap();
-    let state = initialize(config_path.path().to_path_buf());
-    let _app = App::new(state);
+async fn construct_with_no_args() {
```

Comment:
Because I changed the `App::new` signature I had to change these tests too -- so I took the opportunity to rewrite them to use a common fixture to avoid as much repetition and broke them into multiple functions for each different feature